components/engine_rocks/src/import.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use crate::Rocks;
+use engine::rocks::util::prepare_sst_for_ingestion;
+use engine_traits::Import;
+use engine_traits::IngestExternalFileOptions;
+use engine_traits::{Error, Result};
+use rocksdb::set_external_sst_file_global_seq_no;
+use rocksdb::IngestExternalFileOptions as RawIngestExternalFileOptions;
+use std::fs::File;
+use std::path::Path;
+use tikv_util::file::calc_crc32;
+
+pub struct RocksIngestExternalFileOptions(RawIngestExternalFileOptions);
+
+impl IngestExternalFileOptions for RocksIngestExternalFileOptions {
+    fn move_files(&mut self, f: bool) {
+        self.0.move_files(f);
+    }
+}
+
+impl Import for Rocks {
+    type IngestExternalFileOptions = RocksIngestExternalFileOptions;
+
+    fn new_ingest_external_file_options() -> RocksIngestExternalFileOptions {
+        RocksIngestExternalFileOptions(RawIngestExternalFileOptions::new())
+    }
+
+    fn prepare_sst_for_ingestion<P: AsRef<Path>, Q: AsRef<Path>>(
+        &self,
+        path: P,
+        clone: Q,
+    ) -> Result<()> {
+        Ok(prepare_sst_for_ingestion(path, clone).map_err(|e| Error::Other(box_err!(e)))?)
```

Comment:
It looks odd to have `Ok(xxxx.map_err(yyyy)?)`. Either `xxxx.map_err(yyyy)` or `Ok(xxxx?)` should be enough 🤔

Reply:
Done.